lib/icinga/checkable-check.cpp

```diff
@@ -242,6 +242,16 @@ void Checkable::ProcessCheckResult(const CheckResult::Ptr& cr, const MessageOrig
 			OnReachabilityChanged(this, cr, children, origin);
 	}
 
+	if (recovery) {
+		// W/o this children could recover too late, after their problem notifications have been sent.
+		for (auto& child : children) {
+			if (child->GetProblem() && child->GetEnableActiveChecks() && !child->IsLikelyToBeCheckedSoon()) {
```

Comment:
So with `child->IsLikelyToBeCheckedSoon()`, this will return if the child is checked within 60 seconds (unless its check_interval is shorter). Is this enough? Like if you get `check_interval = 2m` you have a 50% change of this triggering or not.

Reply:
And the other 50% result in ref/NC/690718.